daemon/algod/api/server/v2/test/handlers_test.go

```diff
@@ -1941,3 +1940,58 @@ func TestExperimentalCheck(t *testing.T) {
 	require.Equal(t, 200, rec.Code)
 	require.Equal(t, "true\n", string(rec.Body.Bytes()))
 }
+
+func setupTestDevMode(devmode bool, t *testing.T) (v2.Handlers, echo.Context, *httptest.ResponseRecorder, []account.Root, []transactions.SignedTxn, func()) {
+	numAccounts := 1
+	numTransactions := 1
+	offlineAccounts := true
+	mockLedger, rootkeys, _, stxns, releasefunc := testingenv(t, numAccounts, numTransactions, offlineAccounts)
+	mockNode := makeMockNode(mockLedger, t.Name(), nil, cannedStatusReportGolden, devmode)
+	dummyShutdownChan := make(chan struct{})
+	handler := v2.Handlers{
+		Node:     mockNode,
+		Log:      logging.Base(),
+		Shutdown: dummyShutdownChan,
+	}
+	e := echo.New()
+	req := httptest.NewRequest(http.MethodGet, "/", nil)
+	rec := httptest.NewRecorder()
+	c := e.NewContext(req, rec)
+	return handler, c, rec, rootkeys, stxns, releasefunc
+}
+
+func TestTimestampOffsetNotInDevMode(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	handler, c, rec, _, _, releasefunc := setupTestDevMode(false, t)
+	defer releasefunc()
+
+	// TestGetBlockTimeStampOffset 400 - offset is not set and mock node is
+	// not on dev mode
+	err := handler.GetBlockTimeStampOffset(c)
+	require.NoError(t, err)
+	require.Equal(t, 400, rec.Code)
+	// TestSetBlockTimeStampOffset 400 - cannot set timestamp offset when not
+	// in dev mode
+	err = handler.SetBlockTimeStampOffset(c, 1)
+	require.NoError(t, err)
```

Comment:
why is err nil for these tests?

Reply:
I think echo uses an internal error handler that reports the error in the response recorder instead.

I did try to make the error check by adding another check for checking the response body in the latest commit: [6fd3c45](https://github.com/algorand/go-algorand/pull/5296/commits/6fd3c4569adbcb110f3b8c3a42af02b567ba03b9)